extension/options/src/App.js

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from "react";
+
+const defaultOptions = {
+  arrowParens: "avoid",
+  bracketSpacing: true,
+  jsxBracketSameLine: false,
+  printWidth: 80,
+  semi: true,
+  singleQuote: false,
+  tabWidth: 2,
+  trailingComma: "none",
+  useTabs: false
+};
+
+function App() {
+  const [options, setOptions] = useState();
+
+  useEffect(() => {
+    window.chrome.storage.sync.get(defaultOptions, setOptions);
+  }, [setOptions]);
+
+  useEffect(() => {
+    if (options) {
+      window.chrome.storage.sync.set(options);
+    }
+  }, [options]);
+
+  function handleChange({ target: { checked, name, type, value } }) {
+    setOptions({
+      ...options,
+      [name]:
+        type === "checkbox"
+          ? checked
+          : type === "number"
+          ? parseInt(value)
+          : value
+    });
+  }
+
+  return options ? (
```

Comment:
When would `options` evaluate to falsy here?

Reply:
https://github.com/prettier/prettier-chrome-extension/pull/56#discussion_r337854482